src/core/src/sys/systems.rs

```diff
@@ -309,6 +321,33 @@ impl System {
         com / total_mass
     }
 
+    /// Move all particles of a molecule such that the
+    /// molecules center-of-mass position resides
+    /// inside the simulation cell.
+    ///
+    /// # Note
+    ///
+    /// If the `CellShape` is `Infinite` there are no changes
+    /// to the positions.
+    pub fn wrap_molecule(&mut self, molid: usize) {
+        // This catch is not necessary, since for `Infinite`
+        // cells, a wrap will return just the same vector and
+        // hence `delta` will be zero; but it is faster.
+        match self.cell().shape() {
+            CellShape::Infinite => (),
```

Comment:
This is not needed, as the `wrap_vector` function already manage the Infinite cell case. The same code ca be used everywhere.